trio/_threads.py

```diff
@@ -2,23 +2,31 @@
 import queue as stdlib_queue
 from itertools import count
 
+import attr
+
 from . import _core
+from ._sync import CapacityLimiter
 
 __all__ = [
     "current_await_in_trio_thread", "current_run_in_trio_thread",
-    "run_in_worker_thread",
+    "run_in_worker_thread", "current_default_worker_thread_limiter",
 ]
 
 def _await_in_trio_thread_cb(q, afn, args):
+    @_core.disable_ki_protection
+    async def unprotected_afn():
+        return await afn(*args)
     async def await_in_trio_thread_task():
-        nonlocal afn
-        afn = _core.disable_ki_protection(afn)
-        q.put_nowait(await _core.Result.acapture(afn, *args))
+        q.put_nowait(await _core.Result.acapture(unprotected_afn))
     _core.spawn_system_task(await_in_trio_thread_task, name=afn)
 
 def _run_in_trio_thread_cb(q, fn, args):
-    fn = _core.disable_ki_protection(fn)
-    q.put_nowait(_core.Result.capture(fn, *args))
+    @_core.disable_ki_protection
+    def unprotected_fn():
+        return fn(*args)
+    res = _core.Result.capture(unprotected_fn)
+    print(res)
```

Comment:
I wonder if it would make sense to add some off-by-default logging to various parts of trio, hooking into the logging module or somesuch.  Might help with debugging things in general.